File: Assignment/app/src/main/java/com/example/assignment/MainActivity.java
package com.example.assignment;

//import android.provider.AlarmClock.EXTRA_MESSAGE;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
//import android.support.v7.app.AppCompatActivity;
import android.widget.EditText;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private View view;

    public static final String EXTRA_MESSAGE="com.example.Lab3.MESSAGE";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // findViewById(R.id.button).setOnClickListener(new handleButton());
    }


    public void SendMassge(View view){
        Intent i =new Intent(this,MainActivity2.class);
        EditText editText =(EditText) findViewById(R.id.editTextTextPersonName);
        String message=editText.getText().toString();
        i.putExtra(EXTRA_MESSAGE,message);
        startActivity(i);
    }
}

File: GuessThePhrase/app/src/main/java/com/example/guessthephrase/MainActivity.kt
package com.example.guessthephrase

import android.content.ReceiverCallNotAllowedException
import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    lateinit var input : EditText
    lateinit var btn : Button
    lateinit var messages : ArrayList<String>
    lateinit var myCL : ConstraintLayout
    lateinit var rvMain : RecyclerView
    lateinit var txtPromet : TextView
    lateinit var textView : TextView

    private val answer = "Hi there"
    private var myAnswer = ""
    private var myDictionary = mutableMapOf<Int, String>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        myCL = findViewById(R.id.myCL)
        messages = ArrayList()

        rvMain=findViewById<RecyclerView>(R.id.rvMain)
        rvMain.adapter = Message(this , messages)
        rvMain.layoutManager = LinearLayoutManager(this)

        txtPromet = findViewById(R.id.textView)
        textView =findViewById(R.id.textView)
        input = findViewById(R.id.input)
        btn  =findViewById(R.id.btn)

        convertLetterToSatr(answer)
        btn.setOnClickListener {  checkGuess(input.text.toString())}
    }

    private fun convertLetterToSatr(answer:String) {
        for(e in answer.indices){
            if(answer[e].equals(" ")) {
                myDictionary[e] = " "
                myAnswer += " "
            }else{
                myDictionary[e] = "*"
                myAnswer += "*"
            }
        }
        txtPromet.text = myDictionary.toString()
    }
    private fun checkGuess(inpu : String){

        if(inpu.contentEquals(answer) ){
            textView.setTextColor(Color.GREEN)
            textView.text = "You Got it!!"
            messages.add("You guessed $inpu that is ture!")
        }else{
            textView.setTextColor(Color.RED)
            textView.text = "Wrong try again! "

            messages.add("You guessed $inpu and the answer was $answer")
        }
        input.text.clear()
        input.clearFocus()
        rvMain.adapter?.notifyDataSetChanged()
    }
}

